lisa/transformers/rpm_kernel_installer.py

```diff
@@ -44,26 +50,19 @@ def validate(self) -> None:
             )
         runbook: RPMInstallerSchema = self.runbook
         kernel_rpm_path: str = runbook.kernel_rpm_path
-        err: str = f"can not find kernel rpm file: {kernel_rpm_path}"
-        assert os.path.exists(kernel_rpm_path), err
-        assert kernel_rpm_path.endswith(
-            ".rpm"
+        assert self._node.shell.exists(
+            PurePath(kernel_rpm_path)
+        ), f"Node does not contain kernel rpm file: {kernel_rpm_path}"
+        assert self._node.tools[Rpm].is_valid_package(
+            kernel_rpm_path
         ), f"Provided file {kernel_rpm_path} is not an rpm"
 
     def install(self) -> str:
         node = self._node
         runbook: RPMInstallerSchema = self.runbook
         kernel_rpm_path: str = runbook.kernel_rpm_path
 
-        filename = os.path.basename(kernel_rpm_path)
-        remote_path = node.working_path / filename
-        node.shell.copy(
-            PurePath(kernel_rpm_path),
-            remote_path,
-        )
-
         rpm = node.tools[Rpm]
-        rpm.install_local_package(str(remote_path))
-        installed_kernel_version = filename[: -len(".rpm")]
+        rpm.install_local_package(kernel_rpm_path)
```

Comment:
We can continue to assume the file name has the version information. To pass in the version information makes the transformer is harder to use.

Reply:
Older code was like
`installed_kernel_version = filename[: -len(".rpm")]`
I will keep it as it is then. It will assume that files have `.rpm` extension.